dill/logger.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Author: Leonardo Gama (@leogama)
+# Copyright (c) 2022 The Uncertainty Quantification Foundation.
+# License: 3-clause BSD.  The full license text is available at:
+#  - https://github.com/uqfoundation/dill/blob/master/LICENSE
+r"""
+Logging utilities for dill.
+
+Python's logging functionality can be conceptually divided into five steps:
```

Comment:
What's the point of this documentation? (From here to "NOTE")
I'd expect the module docs to cover what logging customization was done, but not cover basic logging. So, what you have under the "customization" header, plus maybe some detail on INFO_DETAIL.  Possibly also any expected formatting for entries to the new `log_trace` method... and an example and any explanation of the details of the printed logger trace that you want to discuss.

Reply:
I agree with you that this documentation of Python's logging framework doesn't fit in the module's documentation. However, I don't think this is "basic logging", but rather "basic logging development". I couldn't find this information in the official documentation, nor in the first resources about logging I've found online, that mostly focused in basic logging usage. I had to untangle the library code to understand that, to do what I wanted to do, I needed to create a `LoggerAdapter` to wrap the `Logger` returned by `logging.getLogger('dill')` (taking care to handle the `process()` method). I just thought that this info should be kept with the implementation. What if this documentation chunk is moved to a comment just above the classes definition? I don't remember if I wrote down this rationale of the classes design, but if not I can add it.

Will complement the remaining as you suggested. 👍🏼

Reply:
> What if this documentation chunk is moved to a comment just above the classes definition?

Perfect